addon/components/layer-result-list.js

```diff
@@ -176,46 +195,90 @@ export default Ember.Component.extend({
     this.set('_noData', false);
     this.set('_displayResults', null);
 
+    let results = this.get('results') || Ember.A();
+
     // If results had been cleared.
-    if (!this.get('results')) {
+    if (Ember.isBlank(results)) {
       return;
     }
 
     this.set('_showLoader', true);
-    let results = this.get('results') || [];
 
     let displayResults = Ember.A();
+
+    // Prepare results format for template.
     results.forEach((result) => {
       let r = {
-        name: Ember.get(result, 'layerModel.name') ? Ember.get(result, 'layerModel.name') : '',
+        name: Ember.get(result, 'layerModel.name') || '',
         settings: Ember.get(result, 'layerModel.settingsAsObject.displaySettings.featuresPropertiesSettings'),
         displayProperties: Ember.get(result, 'layerModel.settingsAsObject.displaySettings.featuresPropertiesSettings.displayProperty'),
+        listForms: Ember.A(),
+        editForms: Ember.A(),
+        features: Ember.A(),
         layerModel: Ember.get(result, 'layerModel')
       };
 
+      let links = Ember.get(result, 'layerModel.layerLink');
+      let layerLink = links.filter(link => link.get('allowShow') === true);
+
+      layerLink.forEach((link) => {
+        if (!Ember.isBlank(link)) {
```

Comment:
Too much nested conditions here and below, they can be simply inverted to reduce the nesting level.